### DISCUSSION TITLE
Trivial File Transfer Protocol (TFTP)

### DISCUSSION THREAD

**Discussion by rob-vangelder** [drakkan/sftpgo#411] [Ideas]
Just wondering if you've considered supporting Trivial File Transfer Protocol (TFTP) as this is often used by network equipment for firmware up/downloads and other kinds of backup/restores.

**Comment by drakkan** [comment]
Hi,

I haven't considered TFTP yet. It is completely different from FTP, it requires a specific implementation. I'm not sure it's worth it

**Comment by the-maldridge** [comment]
Supporting TFTP would be a pretty useful addition in my opinion, though it should probably be off by default due to tftp's more limited concepts of authentication.  Being able to deploy an sftpgo container that I could point both old hardware at that only understands tftp and newer hardware that can SCP its config to a backup server would be a use case that no single solution elegantly solves right now.  Throw in that the data could all be transparently stored on some remote object store and this is a pretty killer feature in my opinion.

And of course by definition the TFTP protocol is, well, trivial.

**Reply by drakkan** [reply]
Hi,

SFTPGo maps users to storage backends based on their credentials, as far as I understand TFTP doesn't support authentication so we should set up storage backend inside config file or find another way to map requests and set restrictions such as permissions, allowed IP etc..

So even if the protocol is trivial and [this](https://github.com/pin/tftp) library could help, its implementation within SFTPGo does not seem so trivial, exposing a new protocol requires a fair amount of work. A [sponsorship](https://github.com/sponsors/drakkan) could help to better understand how we can add support for this feature, thank you

**Reply by the-maldridge** [reply]
To be clear, "trivial" in my original comment referred to the complexity of the protocol itself when compared to something like SFTP.  I still fully expect that anything beyond a read-only implementation would be a non-trivial investment in effort.

**Reply by JonBendtsen** [reply]
> Being able to deploy an sftpgo container that I could point both old hardware at that only understands tftp and newer hardware that can SCP its config to a backup server would be a use case that no single solution elegantly solves right now. Throw in that the data could all be transparently stored on some remote object store and this is a pretty killer feature in my opinion.
> 

I suppose you already found a solution, but if you're already using containers, why not just share a volume between an SFTPGo container and a TFTP container?

Of course data would be stored locally with this option and not remotely
